fsspec/implementations/http.py

```diff
@@ -317,16 +317,19 @@ async def _info(self, url, **kwargs):
         corresponding file will not work).
         """
         size = False
+        kw = self.kwargs.copy()
+        kw.update(kwargs)
         for policy in ["head", "get"]:
             try:
                 session = await self.set_session()
-                size = await _file_size(
-                    url, size_policy=policy, session=session, **self.kwargs
-                )
+                size = await _file_size(url, size_policy=policy, session=session, **kw)
                 if size:
                     break
-            except Exception:
-                pass
+            except Exception as e:
+                import pdb
+
+                pdb.set_trace()
```

Comment:
Nice!

Reply:
Yeah, but it went away very quickly https://github.com/intake/filesystem_spec/commit/64144ed25c33aee4d7b5442847b289a810e5170a

Reply:
I'm not complaining! That is the price we pay for fast implementations :smile:

Reply:
I'm probably supposed to use pycharm or similar to do this kind of thing automatically

Reply:
BTW, I use pre-commits for that. I checked and the one I used only catches the new `breakpoint` syntax. I'm sure there is one for pdb too.